woodwork/datatable.py

```diff
@@ -1021,9 +1024,15 @@ def mutual_information_dict(self, num_bins=10, nrows=None):
             (perfect dependency).
         """
         valid_types = get_valid_mi_types()
+
         valid_columns = [col.name for col in self.columns.values() if (
             col.name != self.index and _get_ltype_class(col.logical_type) in valid_types)]
 
+        if (include_index and
+            self.index is not None and
+                _get_ltype_class(self.columns[self.index].logical_type) in valid_types):
+            valid_columns.append(self.index)
```

Comment:
Another assumption I made: The "index" in question was the DataTable/Woodwork index. 

But now I'm wondering if it might have been in reference to the pandas underlying index. With Woodwork's assumption that if an index is set, it'll match the underlying pandas index, we could change this so that we always add the pandas underlying index as a column for MI and, if Woodwork defined an index, it'd match that. Otherwise, it'd just provide the Pandas index (which could be of note to the user or a RangeIndex if it's not been defined)

Reply:
Would be good to clarify before we close, but I was thinking index referred to Woodwork index, not pandas index...

Reply:
Created a separate issue #697 for using the underlying index if we decide to add this at a later date, but wanted to keep this PR simple, as this wasn't the intended behavior ("index" was, indeed, the Woodwork index)